lib/index.js

```diff
@@ -155,15 +163,50 @@ internals.initialize = (server, next) => {
             return next(err);
         }
 
-        // Maybe run migrations
+        const migrationGroups = collector.knexGroups.map((knexGroup) => {
+
+            const newKnexGroup = Object.assign({}, knexGroup);
+
+            if (!newKnexGroup.knex) {
+                newKnexGroup.knex = rootKnex;
+            }
+
+            // Default to passed-in migrations directory
+
+            if (!newKnexGroup.migrationsDir) {
+                newKnexGroup.migrationsDir = rootMigrationsDir;
+            }
+
+            // Turn these strings into the model classes
+
+            const allModels = server.models(true);
+            newKnexGroup.models = newKnexGroup.models.map((modelName) => {
 
-        if (!collector.migrateOnStart) {
-            return next();
+                return allModels[modelName];
+            });
+
+            return newKnexGroup;
+        });
+
+        // Maybe run schwifty-migration to generate migration file
+
+        if (collector.makeMigrationMode === true) {
+            SchwiftyMigration.genMigrationFile({ migrationGroups }, () => {
```

Comment:
It's more hapi-stylistic (https://eslint.org/docs/rules/no-else-return), but I'd suggest instead returning here and nixing the `else {}` clause.